### PR TITLE
Zoo chroot

### PR DESCRIPTION
**Pull request by arnaudbriche**
Hi,

In our production env, our zookeeper cluster handle multiple kafka clusters, each kafka cluster has it's zk data chrooted to a top-level directory.
The code in zk.go assume that kafka cluster data ils always in the top-level directory, so I just added a ZKChrootPath config in ConsumerConfig and modified zk.go to prepend the chrootPath in each high-level api call.

### CODE REVIEW

File: zk.go [hunk 1]
```diff
@@ -5,13 +5,17 @@ import (
...
 	"time"
 
 	"github.com/samuel/go-zookeeper/zk"
 )
 
 // ZK wraps a zookeeper connection
-type ZK struct{ *zk.Conn }
+type ZK struct {
+	*zk.Conn
+	chrootPath string
```
Comment (dim) [inline zk.go]: I don't think "chroot" is the right term here. I'd call it just `prefix`

File: consumer.go [hunk 2]
```diff
@@ -85,7 +89,8 @@ func NewConsumer(client *sarama.Client, zookeepers []string, group, topic string
 	}
 
 	// Connect to zookeeper
-	zoo, err := NewZK(zookeepers, config.ZookeeperSessionTimeout)
+	zoo, err := NewZK(zookeepers, config.ZKChrootPath, config.ZKSessionTimeout)
```
Comment (dim) [inline consumer.go]: `ZKPrefix` sounds much nicer to me

File: zk.go [hunk 3]
```diff
@@ -21,12 +25,18 @@ type zkConsumerConfig struct {
...
-func NewZK(servers []string, recvTimeout time.Duration) (*ZK, error) {
+func NewZK(servers []string, chrootPath string, recvTimeout time.Duration) (*ZK, error) {
 	conn, _, err := zk.Connect(servers, recvTimeout)
+
 	if err != nil {
 		return nil, err
 	}
-	return &ZK{conn}, nil
+
+	if !strings.HasPrefix(chrootPath, "/") {
```
Comment (dim) [inline zk.go]: Please use:

```go
prefix = path.Clean(prefix)
if !path.Abs(prefix) { 
  prefix = "/" + prefix 
}
```

File: zk.go [hunk 4]
```diff
@@ -21,12 +25,18 @@ type zkConsumerConfig struct {
...
+		chrootPath = "/" + chrootPath
+	}
+
+	return &ZK{Conn: conn, chrootPath: strings.TrimSuffix(chrootPath, "/")}, nil
```
Comment (dim) [inline zk.go]: No need for `TrimSuffix`, once `path.Clean` is used

File: zk.go [hunk 5]
```diff
@@ -21,12 +25,18 @@ type zkConsumerConfig struct {
 }
 
 // NewZK creates a new connection instance
-func NewZK(servers []string, recvTimeout time.Duration) (*ZK, error) {
+func NewZK(servers []string, chrootPath string, recvTimeout time.Duration) (*ZK, error) {
```
Comment (dim) [inline zk.go]: I'd love to see a few more tests for `NewZK` too, e.g. `It("should initialize with path prefix", func() { ... })` with a few prefix examples.

File: zk.go (hunk 1 above)
Comment (arnaudbriche) [inline zk.go]: Well, this is the term they use in the Kafka doc: http://kafka.apache.org/documentation.html

Comment (dim) [inline zk.go]: Oh, right, it's OK then, just sounds a bit strange in my opinion. Maybe that's just me :)

**Comment by dim** [comment]
This has been added in https://github.com/bsm/sarama-cluster/commit/c0b2cb6a7ce766ce74b3093418bab9403d4fa5f8

**Pull request closed without merging**